File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInventoryArmor.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

import java.util.Objects;

public class PSPInventoryArmor implements PlayerStateProperty<ItemStack[]> {
    @Override
    public String getConfigName() {
        return "armor";
    }

    @Override
    public ItemStack[] getDefaultSerializedProperty() {
        return new ItemStack[0];
    }

    @Override
    public ItemStack[] getSerializedPropertyFromPlayer(Player player) {
        return player.getInventory().getArmorContents();
    }

    @Override
    public void restoreProperty(Player player, ItemStack[] armor) {
        player.getInventory().setArmorContents(armor);
    }

    @Override
    @SuppressWarnings("SuspiciousToArrayCall")
    public ItemStack[] getValueFromConfig(ConfigurationSection config) {
        return Objects.requireNonNull(config.getList(getConfigName())).toArray(new ItemStack[0]);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInventoryArmor.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.configuration.ConfigurationSection
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import org.noxet.noxetserver.playerstate.PlayerStateProperty

import scala.jdk.CollectionConverters.*

object PSPInventoryArmor extends PlayerStateProperty[Array[ItemStack]]:
  override def getConfigName: String = "armor"

  override def getDefaultSerializedProperty: Array[ItemStack] = Array.empty

  override def getSerializedPropertyFromPlayer(player: Player): Array[ItemStack] =
    player.getInventory.getArmorContents

  override def restoreProperty(player: Player, armor: Array[ItemStack]): Unit =
    player.getInventory.setArmorContents(armor)

  override def getValueFromConfig(config: ConfigurationSection): Array[ItemStack] =
    config.getList(getConfigName).asScala.toArray